### PR TITLE
i18n - support multiple languages

### PR DESCRIPTION
**Pull request by mapmeld**
I'm starting work on #437 and would like to make sure (because it affects a lot of files) that it looks OK

Here's what it's about:
- create a function ```t``` which works cross-browser for extensions' ```browser.i18n.getMessage``` function, with a rough fallback
- put all UI text phrases into the English locales file, sorted alphabetically

Potential areas for feedback:
- do the names which I created for for UI text phrases look OK?
- should we plan to add descriptions for the remaining UI text phrases? or a few ambiguous ones?
- there is repetition for words which are sometimes "CAPS", "Title Case", or "Sentence case". For example "Cancel" and "CANCEL" buttons. Is it OK to use CSS text-transform on these elements instead?

### CODE REVIEW

File: test/integration/lib/first-time.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ async function runFirstTimeUsageTest(assert, done) {
   // Scroll through terms
   const title = app.find('h1').text()
   // TODO Find where Metamask is getting added twice in the title
-  assert.equal(title, 'MetaMaskMetaMask', 'title screen')
+  assert.equal(title, 'MetaMask', 'title screen')
```
Comment (tmashuang) [inline test/integration/lib/first-time.js]: Change this to
```
const title = app.find('h1')[1]
   // TODO Find where Metamask is getting added twice in the title
assert.equal(title.textContext, 'MetaMask', 'title screen')
```
Should fix the integration test issue.

**Comment by mapmeld** [comment]
@danfinlay @kumavis @tmashuang  I've finished looking for strings in the new UI (this doesn't seem to cover the initial keywords screen?) and removed WIP label from this PR.  I decided to change buttons such as 'SIGN' to 'Sign' plus a CSS class ```allcaps```.

When you run ```npm test```, it runs the fallback function in i18n.js

When you open the extension in one of the browsers, it should use the extension i18n API. When I tested on my computer, it was not enough to add a language to my browser; I had to temporarily change the UI language of my OS to Spanish or Japanese to test that strings were coming from one of the JSON files.

Welcoming your feedback on this

**Comment by mapmeld** [comment]
my future employer (as of tomorrow) may have a conflict with me working on this project, so I am releasing the code of this pull request for others to use

**Comment by danfinlay** [comment]
I'm so sorry we didn't get to review/merge this in time for you, @mapmeld, but thank you so much for the contribution!

**Comment by mapmeld** [comment]
No worries, hope it can be useful!

File: ui/i18n.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+// cross-browser connection to extension i18n API
+
+const chrome = chrome || null
+const browser = browser || null
+let getMessage = null
+
+if ((chrome && chrome.i18n && chrome.i18n.getMessage) ||
+    (browser && browser.i18n && browser.i18n.getMessage)) {
```
Comment (danfinlay) [inline ui/i18n.js]: I would slightly prefer using [extensionizer](https://www.npmjs.com/package/extensionizer) to this inline logic.

File: ui/i18n.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  getMessage = (chrome || browser).i18n.getMessage
+} else {
+  // fallback function
+  console.warn('browser.i18n API not available?')
```
Comment (danfinlay) [inline ui/i18n.js]: We use a global `loglevel` instance. We should document it on our contributing guide. Correct form here will be `log.warn(message)`.

File: ui/i18n.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const msg = require('../app/_locales/en/messages.json')
+  getMessage = function (key, substitutions) {
+    if (!msg[key]) {
+      console.error(key)
```
Comment (danfinlay) [inline ui/i18n.js]: Should be `log.error()` for our global `loglevel` instance.

File: ui/i18n.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+const browser = browser || null
+let getMessage = null
+
+if ((chrome && chrome.i18n && chrome.i18n.getMessage) ||
+    (browser && browser.i18n && browser.i18n.getMessage)) {
+  getMessage = (chrome || browser).i18n.getMessage
+} else {
+  // fallback function
+  console.warn('browser.i18n API not available?')
+  const msg = require('../app/_locales/en/messages.json')
```
Comment (danfinlay) [inline ui/i18n.js]: This looks hard-coded to English. I guess we would need to manually change this and rebuild per language?

Comment (danfinlay) [reply ui/i18n.js]: Oh this is just the fallback function. I'm still looking for where the conditional swap is.

**Review by danfinlay — COMMENTED** [review]
This is a tremendous achievement. It looks pretty fantastic at the code level, left a few notes. To verify this really really works, I'm trying to change my local language and see the language change, but I haven't quite gotten that to work yet.

Has anyone successfully changed their extension language on their Chrome?

**Comment by mapmeld** [comment]
@danfinlay 
- (browser.i18n || chrome.i18n).getMessage picks which locale file to use, this is under the hood, but I believe it is the OS level language. So if you set Chrome to Spanish accept-language, it will not affect the extension. Change the OS settings to Spanish and restart Chrome. Careful to save your place to switch back.
- testing that language UI is working: the strings are all replaced, so if you're seeing English, you can confirm it is running from locales.  Make sure you are looking at the new UI
- testing that language can be switched: if you've confirmed English locale changes affect your UI, then make an entry for a language in app/_locales.  I had to add a Japanese name and switch the language on the OS level for this to show up in Chrome

**Comment by danfinlay** [comment]
I implemented some changes based on my review, which also got it working on my machine (Mac). GitHub won't let me open that PR unless I close this one first (same branch name?).

**Pull request closed without merging**